### ISSUE TITLE
Unwanted formatting in Latex in $ $ and $$ $$ in Markdown 

### ISSUE THREAD

**Issue by nialltc** [prettier/prettier#5197]
**Prettier 1.14.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEASVBDA+sAzgVwCNcwAnASwAcYBfdAHSkxwOLKttUfQEEAqAEIMm-ISAA0ICNXLRcyUBlKkIAdwAKShPJQYAbhHIATCSEKkMYANZwYAZUqXyUAObIYpfHEnPccUjDqFi4AthjIAGYYADZ+kgBWuAAeAhbWtnYYIXAAMs5wkTFxII6kfqTIIGGkVkZqUKaUFLAA6sYwABbIABwADJJNEH4tFpSVTXDlegWSpHAAjvjkc0EYoeFIUbHeIH4h5O6eO7jOLtFwAIr4EPCF25IwGIRtRp3IAEwPFuTRpwDCEBCYUqUGgMxA+D8ABUnjotn4aDQgA)
```sh
--parser markdown
```

**Input:**
```markdown
$$a_{subscript}$$
$a_{subscript}$
$$A*B$$
$A*B$
```

**Output:**
```markdown
$$a*{subscript}$$
$a*{subscript}$
$$A*B$$
$A*B$

```

On my computer, (prettier 1.14.3, MacOS through prettier-atom) it also does this but could not replicate on playground:

**Input:**
```markdown
$$A*B$$
$A*B$
```

**Output:**
```markdown
$$A\*B$$
$A\*B$

```

**Expected behavior:**
```markdown
$$a_{subscript}$$
$a_{subscript}$
$$A*B$$
$A*B$
```

**Comment by ikatyang** [comment]
Hi, this should be fixed by #5050. If you'd like to try it out now, you could install it from GitHub (`npm install prettier/prettier`).

**Prettier pr-5050**
[Playground link](https://deploy-preview-5050--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEASVBDA+sAzgVwCNcwAnASwAcYBfdAHSkxwOLKttUfQEEAqAEIMm-ISAA0ICNXLRcyUBlKkIAdwAKShPJQYAbhHIATCSEKkMYANZwYAZUqXyUAObIYpfHEnPccUjDqFi4AthjIAGYYADZ+kgBWuAAeAhbWtnYYIXAAMs5wkTFxII6kfqTIIGGkVkZqUKaUFLAA6sYwABbIABwADJJNEH4tFpSVTXDlegWSpHAAjvjkc0EYoeFIUbHeIH4h5O6eO7jOLtFwAIr4EPCF25IwGIRtRp3IAEwPFuTRpwDCEBCYUqUGgMxA+D8ABUnjotsUMLgYHdinUwCidhB8DBKNj3hiaDQgA)
```sh
--parser markdown
```

**Input:**
```markdown
$$a_{subscript}$$
$a_{subscript}$
$$A*B$$
$A*B$
```

**Output:**
```markdown
$a_{subscript}$
$a_{subscript}$
$A*B$
$A*B$

```

**Comment by ikatyang** [comment]
Duplicate of #3700
